feat(a2a): add delete functionality / event links

dwmkerr opened the pull request:
Delete A2A server, show availability, direct link to events. Brings a2a server view in line with other pages.

- Adds delete button with confirmation dialog for A2A servers

<img width="1134" height="397" alt="image" src="https://github.com/user-attachments/assets/ad7c96fe-63e4-4fbe-8ab7-d09a80364098" />


- Replaces StatusBadge with AvailabilityStatusBadge that links to event logs：
<img width="508" height="202" alt="image" src="https://github.com/user-attachments/assets/0746d953-a3f6-4419-a4d3-069582c93b43" />

<img width="1459" height="437" alt="image" src="https://github.com/user-attachments/assets/35983e4e-2ab3-4661-b18e-0351b40da4c8" />


- Implements consistent UI patterns matching Models and Agents resources

Fixes #222

dwmkerr committed:
feat: add delete functionality and event links for A2A servers

Implements issue #222 by adding:
- Delete button with confirmation dialog for A2A servers (both card and row views)
- AvailabilityStatusBadge with clickable event links
- Consistent UI patterns matching Models and Agents resources

Changes:
- Updated A2AServerCard to include delete action and AvailabilityStatusBadge
- Updated A2AServerRow to include delete button and AvailabilityStatusBadge
- Updated A2AServersSection to handle delete operations
- Status badges now link to /events page filtered by A2AServer kind

dwmkerr committed:
fix: map A2A server name to id field for delete operations

The API returns 'name' but the UI expects 'id'. Now mapping name to id
in all service methods to ensure delete and other operations work correctly.

Fixes issue where delete was sending 'undefined' as the identifier.

dwmkerr committed:
Merge branch 'main' into feat/a2a-server-delete-and-events

dwmkerr committed:
Merge branch 'main' into feat/a2a-server-delete-and-events

dwmkerr committed:
Merge branch 'main' into feat/a2a-server-delete-and-events

dwmkerr committed:
test: add comprehensive tests for A2A server delete and availability features

Adds unit tests for new functionality in PR #436:
- Service layer tests for A2AServersService delete, create, update methods
- Component tests for A2AServerCard delete button and confirmation dialog
- Component tests for A2AServerRow delete and info actions
- Component tests for A2AServersSection lifecycle and error handling

Coverage for a2a-servers.ts increased to 65.95%

daniele-marostica commented on services/ark-dashboard/ark-dashboard/__tests__/unit/components/cards/a2a-server-card.test.tsx:
```diff
@@ -0,0 +1,214 @@
+      />
+    );
+
+    expect(screen.getByLabelText('Delete a2a server')).toBeInTheDocument();
```
This is fine as this component only mentions "delete" in this scope. For better matching consider getting element by [aria role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Guides/Techniques), eventually specifying the name of the element

```suggestion
    expect(screen.getByRole('button', { name: /delete/i })).toBeInTheDocument();
```

dwmkerr replied:
done!

daniele-marostica commented on services/ark-dashboard/ark-dashboard/__tests__/unit/components/cards/a2a-server-card.test.tsx:
```diff
@@ -0,0 +1,214 @@
+      />
+    );
+
+    fireEvent.click(screen.getByLabelText('Delete a2a server'));
```
consider using the `userEvent` API instead of the lower level `fireEvent`

[here the differences between these two
](https://testing-library.com/docs/user-event/intro/#difference-to-fireevent)

```suggestion
    userEvent.click(screen.getByRole('button', { name: /delete/i }));
```

dwmkerr replied:
done

dwmkerr committed:
fix: use getByRole and userEvent in A2A server card tests

Address PR review feedback:
- Replace getByLabelText with getByRole for better ARIA role matching
- Replace fireEvent with userEvent for more realistic interactions

dwmkerr committed:
Merge branch 'main' into feat/a2a-server-delete-and-events

dwmkerr committed:
fix: replace fireEvent with userEvent in A2A server tests

dwmkerr commented:
Merging w/ approval from @daniele-marostica
